ToricVarieties/makefile

```diff
@@ -12,7 +12,8 @@ docclean:
 	(cd doc ; ./clean)
 
 test:	doc
-	gap maketest.g
+	(cd tst ; ./clean)
+	gap tst/testall.g
 
 test-with-coverage:	doc
 	OUTPUT=$$(/usr/bin/time --quiet --format="%U %S\n%e" --output=performance.out gap --banner --quitonbreak --cover stats maketest.g 2>&1); \
```

Comment:
You have to adjust this, too. Luckily, the `extract_examples` method exits with a correct exit code, so you should be able to replace this line and the three following lines simply by
```
/usr/bin/time --quiet --format="%U %S\n%e" --output=performance.out gap --quitonbreak --cover stats tst/testall.g
```

Reply:
Done in https://github.com/homalg-project/ToricVarieties_project/pull/49/commits/5c5d4ad71fe18bc321a45a9669c6d02640b5d836.

Reply:
Your change might work, but my suggestion was to get rid of the `grep` stuff, i.e., replace all three lines by the line I suggested above.

Reply:
Sorry - my mistake. I have just corrected this. Now the ci-test completes successfully, but make test still fails due to both the singular and the IO_ForHomalg banner. Note that I am loading IO_ForHomalg at the beginning of this example, i.e. before the actual test. Also, I am suppressing banners in this example file.

Any suggestion on how to fix make test? Is there a strong reason for having a separate ci-test and an "ordinary" test? If not, how about either remvoing make test or making it point to the target ci-test?

Reply:
> I have just corrected this in 367c26b.

Your line still differs from my suggestion. Now `make ci-test` never shows any output, because you capture it in the variable `OUTPUT`. And more important, it also never fails because the trailing backslash lets `make` use the return value of the call in the next line, which always succeeds (if GAP is installed).

> Now the ci-test completes successfully, but make test still fails

If `make test` fails, `make ci-test` must also fail, otherwise the CI does not catch all errors. As mentioned above: Please test your changes by simulating a failing test and executing `make test` and `make ci-test`, so we can be sure, errors are actually caught in both cases.

> Is there a strong reason for having a separate ci-test and an "ordinary" test?

As you can see, `make ci-test` collects coverage and performance data, which increases the runtime of the test. Since one usually does not need coverage and performance data when running the tests manually, we have separate targets. And for example in `FinSetsForCAP`, `make ci-test` even runs some additional tests.

Reply:
Sorry - I am silly. Have a look at https://github.com/homalg-project/ToricVarieties_project/pull/49/commits/f4dc577bd2c999f8eaabd18744392324ee2b7341. This (hopefully) not matches your suggestion.

Also, I correct my statement. The banner error is still there for both tests.

Reply:
> This (hopefully) matches your suggestion.

Technically your version still differs by the use of the `--banner` option. This is not a problem but makes debugging harder if a package failes to load.

> The banner error is still there for both tests.

I think I found the problem: Everything you added in 40205b47230b3120accf2a391cd83b1e2927e01e is outside of the `@Example` block and thus ignored by `AutoDoc`.

Reply:
Thank you for the pointer. I have worked this in right now:
* With these changes, there are no fails for both make test and make ci-test.
* Also, I have verified, that these test catch fails (e.g. 1+1 = 3 is detected).

Further changes required?